performant-software/basira
# BASIRA #129 - Convert Location Role to Value List

### What this PR does

- Per #129:
  - Converts Location Role into a Value List when editing Artwork, Place, or Person
  - Makes Location a Qualifiable type of record and updates its serialization
  - Makes `getValueListValue` usable by any Qualifiable type

### Status


### Steps to test

1. Visit https://basiraproject-staging.herokuapp.com/
2. Log in
3. Click Value Lists
4. Go to Location and add a new value with Object = "Location" and Group = "Role", type whatever for the human name
5. Go back to the Artworks list
6. Click Edit on an artwork
7. Click Locations
8. Click Edit next to a location if there are any, otherwise add one
9. Verify that the "Role" list in the dialog that appears is a dropdown, and that you can choose the value you entered in step 4
10. Choose a value and click save, and then save the artwork record
11. Refresh and verify that the Role value persisted for the location
12. Click on the location's name to go to its page
13. Go to Artworks in the location edit page
14. Click Edit next to the attached artwork
15. Verify that its Role dropdown has the correct value
